### ISSUE TITLE
Disable plugin error when plugin manager was set to one other than 'dein'

### ISSUE THREAD

**Issue by hongyuanjia** [wsdjeg/SpaceVim#1354]
## Expected behavior, english is recommended

`g:spacevim_disabled_plugins` only available when plugin manager was set to `'dein'` as it directly calls `dein#disable`. Errors will occur when plugin manager is set to `'vim-plug'` or `'neobundle'`.

What's more, on Windows, you have to manually download `vim-plug` and save it to `~/.cache/vim-plug/autoload/plug.vim`, as SpaceVim tries to download that file using `curl` which is unavailable on Windows. `vim-plug` has provided install script using PowerShell for Windows users, and it is recommended to include that method in SpaceVim. I adopted it according paths of SpaceVim:

```PowerShell
# adopted from https://github.com/junegunn/vim-plug/blob/master/README.md
md ~\.cache\vim-plug\autoload
$uri = 'https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim'
(New-Object Net.WebClient).DownloadFile(
  $uri,
  $ExecutionContext.SessionState.Path.GetUnresolvedProviderPathFromPSPath(
    "~\.cache\vim-plug\autoload\plug.vim"
  )
)
```

## Environment Information
- OS: Windows 10
- vim version: 8.0
- neovim version: NA

## The reproduce ways from Vim starting (Required!)

Just add the line below in `init.vim` and restart vim.

```viml
let g:spacevim_plugin_manager = 'vim-plug'
```

### SpaceVim Options :

```viml
g:spacevim_statusline_left_sections = ['winnr', 'filename', 'major mode', 'version control info']
g:spacevim_enable_statusline_display_mode = 0
g:spacevim_enable_os_fileformat_icon = 1
g:spacevim_leader_guide_position = 'botright'
g:spacevim_version = '0.7.0-dev'
g:spacevim_force_global_config = 0
g:spacevim_smartcloseignoreft = ['help', 'tagbar', 'vimfiler', 'SpaceVimRunner', 'SpaceVimREPL', 'SpaceVimQuickFix', 'HelpDescribe', 'VebuggerShell', 'VebuggerTerminal']
g:spacevim_default_indent = 4
g:spacevim_enable_key_frequency = 0
g:spacevim_enable_javacomplete2_py = 0
g:spacevim_enable_powerline_fonts = 1
g:spacevim_enable_language_specific_leader = 1
g:spacevim_warning_symbol = '⚠'
g:spacevim_simple_mode = 0
g:spacevim_colorscheme = 'molokai'
g:spacevim_enable_debug = 1
g:spacevim_enable_guicolors = 1
g:spacevim_custom_plugins = [['jalvesaq/Nvim-R'], ['vim-pandoc/vim-pandoc'], ['vim-pandoc/vim-pandoc-syntax'], ['vim-pandoc/vim-rmarkdown'], ['ntpeters/vim-better-whitespace', {'on_cmd': 'EnableStripWhitespaceOnSave'}], ['kana/vim-textobj-fold'], ['Julian/vim-textobj-variable-segment'], ['beloglazov/vim-textobj-quotes'], ['sgur/vim-textobj-parameter'], ['terryma/vim-expand-region'], ['thinca/vim-qfreplace']]
g:spacevim_leader_guide_submode_mappings = {'<C-C>': 'win_close'}
g:spacevim_buffer_index_type = 4
g:spacevim_hiddenfileinfo = 1
g:spacevim_statusline_inactive_separator = 'arrow'
g:spacevim_gitcommit_pr_icon = ''
g:spacevim_plugin_manager = 'vim-plug'
g:spacevim_error_symbol = '✖'
g:spacevim_vim_plug_installed = 1
g:spacevim_windows_smartclose = 'q'
g:spacevim_windows_leader = 's'
g:spacevim_wildignore = '*/tmp/*,*.so,*.swp,*.zip,*.class,tags,*.jpg,*.ttf,*.TTF,*.png,*/target/*,.git,.svn,.hg,.DS_Store,*.svg'
g:spacevim_use_colorscheme = 1
g:spacevim_enable_ale = 0
g:spacevim_enable_insert_leader = 1
g:spacevim_enable_cursorcolumn = 0
g:spacevim_colorscheme_bg = 'dark'
g:spacevim_enable_neocomplcache = 0
g:spacevim_leader_guide_sort_horizontal = 0
g:spacevim_enable_vimfiler_welcome = 0
g:spacevim_max_column = 80
g:spacevim_auto_disable_touchpad = 1
g:spacevim_vim_help_language = 'en'
g:spacevim_gitcommit_issue_icon = ''
g:spacevim_lint_on_save = 1
g:spacevim_search_tools = ['rg', 'ag', 'pt', 'ack', 'grep']
g:spacevim_dein_installed = 0
g:spacevim_autocomplete_method = 'deoplete'
g:spacevim_enable_vimfiler_filetypeicon = 0
g:spacevim_plugin_manager_max_processes = 8
g:spacevim_statusline_unicode_symbols = 1
g:spacevim_statusline_right_sections = ['fileformat', 'cursorpos', 'percentage']
g:spacevim_unite_leader = 'f'
g:spacevim_enable_ycm = 0
g:spacevim_enable_tabline_filetype_icon = 1
g:spacevim_colorscheme_default = 'desert'
g:spacevim_plugin_groups_exclude = []
g:spacevim_statusline_separator = 'arrow'
g:spacevim_project_rooter_automatically = 1
g:spacevim_neobundle_installed = 0
g:spacevim_disabled_plugins = ['deoplete.vim']
g:spacevim_leader_guide_max_size = 0
g:spacevim_guifont = ''
g:spacevim_info_symbol = 'ⓘ'
g:spacevim_plugin_groups = ['web', 'lang', 'edit', 'ui', 'tools', 'checkers', 'format', 'chat', 'git', 'VersionControl', 'javascript', 'ruby', 'python', 'scala', 'lang#go', 'lang#markdown', 'scm', 'editing', 'indents', 'navigation', 'misc', 'core', 'core#banner', 'core#statusline', 'core#tabline', 'default', 'unite', 'github', 'denite', 'ctrlp', 'autocomplete', 'vim', 'windows', 'incsearch', 'lang#python', 'lang#vim', 'shell', 'tools#screensaver', 'colorscheme']
g:spacevim_leader_guide_flatten = 1
g:spacevim_sidebar_width = 30
g:spacevim_enable_googlesuggest = 0
g:spacevim_enable_neomake = 1
g:spacevim_plugin_bundle_dir = 'C:\Users\hongy\.cache\vimfiles\'
g:spacevim_leader_guide_run_map_on_popup = 1
g:spacevim_hosts_url = 'https://raw.githubusercontent.com/racaljk/hosts/master/hosts'
g:spacevim_relativenumber = 1
g:spacevim_leader_guide_default_group_name = ''
g:spacevim_smartcloseignorewin = ['__Tagbar__', 'vimfiler:default']
g:spacevim_terminal_cursor_shape = 2
g:spacevim_altmoveignoreft = ['Tagbar', 'vimfiler']
g:spacevim_keep_server_alive = 1
g:spacevim_snippet_engine = 'ultisnips'
g:spacevim_project_rooter_patterns = ['.git/', '_darcs/', '.hg/', '.bzr/', '.svn/']
g:spacevim_realtime_leader_guide = 1
g:spacevim_github_username = 'hongyuanjia'
g:spacevim_leader_guide_hspace = 5
g:spacevim_denite_leader = 'F'
g:spacevim_checkinstall = 1
g:spacevim_windows_index_type = 3
g:spacevim_filetype_icons = {}
g:spacevim_plugin_name = 'vim-qfreplace'
g:spacevim_enable_cursorline = 1
g:spacevim_custom_color_palette = []
g:spacevim_lint_on_the_fly = 0
g:spacevim_debug_level = 1
g:spacevim_filemanager = 'vimfiler'
g:spacevim_src_root = 'E:\sources\'
g:spacevim_leader_guide_vertical = 0
g:spacevim_language = ''
g:spacevim_enable_vimfiler_gitstatus = 0
```

### SpaceVim Health checking :

```
SpaceVim clipboard support check report:
Checking +clipboard:
      SUCCEED!
SpaceVim environment check report:
Current progpath: gvim.exe(C:\Program Files\Vim\vim80\gvim.exe)
version: 800
OS: Windows
[shell, shellcmdflag, shellslash]: ['C:\WINDOWS\system32\cmd.exe', '/c', 0]
SpaceVim lua support check report:
Checking +lua:
      SUCCEED!
SpaceVim python support check report:
Checking +python3:
      SUCCEED!
Checking +python:
      SUCCEED!
```

### SpaceVim runtime log :

```log
[ SpaceVim ] : logger file  does not exists, only log for current process will be shown!
[ SpaceVim ] [12:55:39] [ Info ] Find project root('.git/','C:\Users\hongy\.SpaceVim.d\init.vim'):C:\Users\hongy\.SpaceVim.d\.git\
[ SpaceVim ] [12:55:39] [ Info ] change to root:C:\Users\hongy\.SpaceVim.d
```

**Comment by wsdjeg** [comment]
@hongyuanjia Thanks for report this info, as We do not use vim-plug as default plugin manager, it is easy for me to find this bug.  BTW if you are using vim-plug, many plugin config can not be loaded, because it do not provide plugin hooks.

**Comment by hongyuanjia** [comment]
@wsdjeg Yes, you are right. After changed to `vim-plug`, most plugins are not loaded. But if I use `dein`, everything works perfectly.

**Comment by wsdjeg** [comment]
Sorry, I can not fix this issue, as vim-plug does not support the feature we need. 

BTW, in spacevim, we have create a UI for dein, and why switch to vim-plug?

**Comment by hongyuanjia** [comment]
Before I switched to SpaceVim, I used `vim-plug`. So, no big deal, just want to preserve all my familiar stuff. SpaceVim is really brilliant.

BTW, if this issue cannot be fixed, it is recommended to modify section ["Custom Configuration"](http://spacevim.org/documentation/) in the documentation:

```vimL
" Set plugin manager, you want to use, default is dein.vim
let g:spacevim_plugin_manager = 'dein'  " neobundle or dein or vim-plug
```

As at current, `'dein'` is the only option in order to make all features of SpaceVim available.
